aws_lambda_powertools/logging/types.py

```diff
@@ -0,0 +1,10 @@
+from typing import Any, Dict, Iterable, Union
+
```

Comment:
```suggestion
from __future__ import annotations
```

This will gives us a nicer syntax for Unions among other things. For example:

* `timestamp: Union[str, int]` becomes `timestamp: str | int`

If you'd like to learn more about this topic, check this out on how this enables Postponed Evaluation of Annotations (or type hint)

https://peps.python.org/pep-0563/

:)

Reply:
Out of curiosity, is there a reason why this isn't used for some of the other typing hints, like for return values on functions?

Reply:
You can definitely use for return values. There's generally two gotchas with this feature:

1. [Reflections will see the annotated type as string until runtime](https://peps.python.org/pep-0649/). This means an extra step to get a forward reference (-> Response is now -> "Response").
2. IDEs will show '|' as a regular `Union` as `|` it's merely syntactic sugar. Eventually, it's easy to go overboard with potentially multiple function returns and it can degrade the caller experience, as it now has to make multiple checks before proceeding.

Hopefully that makes sense, let me know if I misunderstood your question otherwise.